specifyweb/frontend/js_src/lib/components/Header/NotificationRenderers.tsx

```diff
@@ -119,6 +124,67 @@ export const notificationRenderers: IR<
       />
     );
   },
+  'record-merge-starting'(notification) {
+    const tableName = notification.payload.table;
+    const collectionId = parseInt(notification.payload.collection_id);
+    const mergeName = notification.payload.name;
+    const schemaResource = new schema.models.Collection.Resource({
+      id: collectionId,
+    });
+    const [collectionName, setCollectionName] = React.useState<string | null>(
+      ''
+    );
+    schemaResource
```

Comment:
did you forget to wrap this in a useEffect?

you are creating a new collection resource on each render, and then making a network request on each render, and that network request modifies the state. That's a recipe for a terrible performance or an infinite loop!

fetching data outside of useEffect is not allowed (though, recently added react server components would allow for this, but only on the back-end)
you don't know how many times the render function will get called

ideally, the render method should not have any side effects at all